tests/Integration/Behaviour/Features/Context/Domain/ShipmentFeatureContext.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use Cart;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Query\GetOrderShipments;
+use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+
+class ShipmentFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @Then the order :orderReference should have the following shipments:
+     *
+     * @param string $orderReference
+     * @param TableNode $table
+     *
+     * @throws RuntimeException
+     */
+    public function verifyOrderShipment(string $orderReference, TableNode $table)
+    {
+        $data = $table->getRowsHash();
+        $orderId = SharedStorage::getStorage()->get($orderReference);
```

Comment:
Little tip, we have some helper that help doing this It's not mandatory but it helps with the readability of the Context code

```suggestion
        $orderId = $this->referenceToId($orderReference);
```

We also have a similar one for list of references `$this->referencesToIds` that is very convenient Both methods are defined on `AbstractDomainFeatureContext`